integrations/waffle-mars.md

```diff
@@ -0,0 +1,402 @@
+}
+```
+
+7. You'll also need to add a script in the `package.json` to run your tests:
+```json
+"scripts": {
+  "build": "waffle",
+  "test": "mocha",
+},
+```
+
+8. You're all set to run the tests, simply use the `test` script you just created and run:
+```
+npm run test
+```
+Please note that it could take a few minutes to process, but if all worked as expected, you should have one failing test.
+
+9. Next, you can go back and edit the test to check for 10 tokens:
+```typescript
+it('Mints the correct initial balance', async () => {
+  expect(await token.balanceOf(wallet.address)).to.equal(10); // This should pass
+});
+```
+10. If you run the tests again, you should now see one passing test:
+```
+npm run test
+```
+
+11. You've tested the ability to mint tokens, next you'll test the ability to transfer the minted tokens. If you want to write a failing test first again that is up to, however the final test should look like this:
+```typescript
+it('Should transfer the correct amount of tokens to the destination account', async () => {
+  // Send the destination wallet 7 tokens
+  await (await token.transfer(walletTo.address, 7)).wait();
+
+  // Expect the destination wallet to have received the 7 tokens
+  expect(await token.balanceOf(walletTo.address)).to.equal(7);
+});
+```
+
+Congratulations, you should now have two passing tests! Altogether, your test file should look like this:
+```typescript
+import { use, expect } from 'chai';
+import { Provider } from '@ethersproject/providers';
+import { solidity } from 'ethereum-waffle';
+import { ethers, Wallet } from 'ethers';
+import { MyToken, MyTokenFactory } from '../build/types';
+
+use(solidity);
+
+describe ('MyToken', () => {
+  let provider: Provider = new ethers.providers.JsonRpcProvider('https://rpc.testnet.moonbeam.network');
+  let wallet: Wallet;
+  let walletTo: Wallet;
+  let token: MyToken;
+
+  beforeEach(async () => {
+    const PRIVATE_KEY = '<insert-your-private-key-here>'
+    wallet = new Wallet(PRIVATE_KEY).connect(provider);
+    walletTo = Wallet.createRandom().connect(provider);
+    token = await new MyTokenFactory(wallet).deploy();
+    let contractTransaction = await token.initialize(10);
+    await contractTransaction.wait();
+  });
+
+  it('Mints the correct initial balance', async () => {
+    expect(await token.balanceOf(wallet.address)).to.equal(10);
+  });
+
+  it('Should transfer the correct amount of tokens to the destination account', async () => {
+    await (await token.transfer(walletTo.address, 7)).wait();
+    expect(await token.balanceOf(walletTo.address)).to.equal(7);
+  });
+})
+```
+
+If you want to write more tests on your own, you could consider testing transfers from accounts without any funds or transfers from accounts without enough funds.
+
+You've used Waffle to write, compile, and test the `MyToken` contract and can now move on to deploying the contract to Moonbase Alpha with Mars. 
```

Comment:
It is actually funny that the contract has been already deployed to Moonbase Alpha for testing right?

Reply:
not with mars but yes technically it was deployed for testing purposes